providers/shared/components/directory/id.ftl

```diff
@@ -0,0 +1,106 @@
+[#ftl]
+
+[@addComponentDeployment
+    type=DIRECTORY_COMPONENT_TYPE
+    defaultGroup="solution"
+/]
+
+[@addComponent
+    type=DIRECTORY_COMPONENT_TYPE
+    properties=
+        [
+            {
+                "Type"  : "Description",
+                "Value" : "A managed directory services instance"
+            }
+        ]
+    attributes=
+        [
+            {
+                "Names" : "Engine",
+                "Types" : STRING_TYPE,
+                "Values" : [ "Simple", "ActiveDirectory" ],
+                "Mandatory" : true
+            },
+            {
+                "Names" : [ "Certificate", "Hostname" ],
```

Comment:
Can remove the Certificate option here and maybe just use Hostname. Also worth including a description that this is for the Domain FQDN

Reply:
Done